packages/cli/scripts/start.js

```diff
@@ -693,18 +689,19 @@ function handleRunTypeCheck(srcPath, inputFiles, tsconfig, config) {
 }
 
 async function getHandlerFilePath(appPath, srcPath, handler) {
-  const parts = handler.split(".");
+  const parts = path.basename(handler).split(".");
```

Comment:
It doesn't handle this case, as suggested here - https://github.com/serverless-stack/serverless-stack/issues/600
```
handler: 'src/users/index.controller.main'
```

We actually do something similar on `sst deploy`. Let's re-use the same logic - https://github.com/serverless-stack/serverless-stack/blob/master/packages/resources/src/util/nodeBuilder.ts#L130